packages/react/src/primitives/shared/utils.ts

```diff
@@ -193,3 +196,27 @@ export const classNameModifierByFlag = (
 ): string => {
   return flag ? `${base}--${modifier}` : null;
 };
+
+export const getCSSVariableIfValueIsThemeKey = (
+  propKey: string,
+  value: string | number
```

Comment:
Is it possible for value to be `null`, `undefined` or `boolean`?

Reply:
Nil value has been filter out - https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/shared/styleUtils.ts#L130

Reply:
For `boolean`, I don't if people would supply a boolean to props because they will get syntax errors. But we can filter that out as well